pkg/controller/util/cache.go

```diff
@@ -0,0 +1,22 @@
+package util
+
+import (
+	"k8s.io/client-go/tools/cache"
+	toolscache "sigs.k8s.io/controller-runtime/pkg/cache"
+)
+
+func GetCacheIndexer(informer toolscache.Informer) cache.Indexer {
+	switch informer.(type) {
+	case cache.SharedIndexInformer:
+		return informer.(cache.SharedIndexInformer).GetIndexer()
+	case toolscache.Informer:
```

Comment:
This code will always match the `toolscache.Informer` case, because the function param type is `toolscache.Informer`.

Reply:
done